src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/forcemerge/WaitForForceMergeStep.kt

```diff
@@ -0,0 +1,145 @@
+
+        // If there are no longer shardsStillMergingSegments, then the force merge has completed
+        if (shardsStillMergingSegments == 0) {
+            logger.info("Force merge completed on [$indexName]")
+
+            stepStatus = StepStatus.COMPLETED
+            info = mapOf("message" to "Force merge completed")
+        } else {
+            /*
+             * If there are still shards with segments merging then no action is taken and the step will be reevaluated
+             * on the next run of the ManagedIndexRunner.
+             *
+             * However, if an internal timeout of 12 hours is reached, the force_merge action will be marked as failed.
+             * This is for cases where the force merge stops silently, such as when nodes relocate during a force
+             * merge operation. Without a timeout, the segment count would stop going down as merging would no longer
+             * occur and the managed index would become stuck in this action.
+             */
+            val timeWaitingForForceMerge: Duration = Duration.between(getStepStartTime(), Instant.now())
+            if (timeWaitingForForceMerge.toSeconds() > FORCE_MERGE_TIMEOUT_IN_SECONDS) {
+                logger.error(
+                    "Force merge on [$indexName] timed out with [$shardsStillMergingSegments] shards containing unmerged segments"
+                )
+
+                stepStatus = StepStatus.FAILED
+                info = mapOf("message" to "Force merge timed out")
+            } else {
+                logger.debug(
+                    "Force merge still running on [$indexName] with [$shardsStillMergingSegments] shards containing unmerged segments"
+                )
+
+                stepStatus = StepStatus.CONDITION_NOT_MET
+                info = mapOf("message" to "Waiting for force merge to complete")
+            }
+        }
+    }
+
+    private fun getMaxNumSegments(): Int? {
+        val actionProperties = managedIndexMetaData.actionMetaData?.actionProperties
+
+        return if (actionProperties != null) {
+            actionProperties.getInt(ActionProperties.MAX_NUM_SEGMENTS)
+        } else {
+            stepStatus = StepStatus.FAILED
+            info = mapOf("message" to "Unable to retrieve [${ActionProperties.MAX_NUM_SEGMENTS}] from ActionProperties=$actionProperties")
+            null
+        }
+    }
+
+    private suspend fun getShardsStillMergingSegments(indexName: String, maxNumSegments: Int): Int? {
+        try {
+            val statsRequest = IndicesStatsRequest().indices(indexName)
+            val statsResponse: IndicesStatsResponse = client.admin().indices().suspendUntil { stats(statsRequest, it) }
+
+            if (statsResponse.status == RestStatus.OK) {
+                return statsResponse.shards.count { it.stats.segments.count > maxNumSegments }
+            }
+
+            stepStatus = StepStatus.FAILED
+            info = mapOf(
+                "message" to "Failed to get index stats",
```

Comment:
This is returning a message to the customer about an internal request we're using in a step, let's change the message to something about the force merge action instead.

Reply:
Updated the messages about ```"Failed to get index stats"``` in this Step as well as ```AttemptRolloverStep``` and ```AttemptTransitionStep```